command/agent.go

```diff
@@ -862,12 +860,11 @@ func (c *AgentCommand) Run(args []string) int {
 	if err := c.storePidFile(config.PidFile); err != nil {
 		c.UI.Error(fmt.Sprintf("Error storing PID: %s", err))
 		return 1
-	} else {
-		if notifyErr := c.NotifierSystemd.Notify(systemd.Ready); notifyErr != nil {
-			c.logger.Debug("server: systemd notify failed", notifyErr)
-		}
 	}
 
+	// Notify systemd that the server is ready (if applicable)
+	c.notifySystemd(systemd.SdNotifyReady)
```

Comment:
I tried adding this a bit later after the run condition which may be more valid to NotifyReady as and when it's actually ready - however since the time it may take for that is not as easily determinable then getting an earlier start notification coupled with a followed up by any stop is probably better than just getting NotifyStopping without ever seeing NotifyReady